### ISSUE TITLE
Uncaught TypeError: Argument 1 passed to PDepend\Source

### ISSUE THREAD

**Issue by JeroenDeDauw** [pdepend/pdepend#258]
```
git clone https://github.com/wmde/FundraisingFrontend.git
cd FundraisingFrontend
git checkout 89a75a4aa02d2457cdb70f796167423011b63767
composer install
php -v
vendor/bin/phpmd --version
```

> PHP 7.0.4-7ubuntu2 (cli) ( NTS )
> PHPMD 2.4.3

```
vendor/bin/phpmd tests/ text phpmd.xml
```

```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to PDepend\Source\AST\AbstractASTNode::addChild() must implement interface PDepend\Source\AST\ASTNode, instance of PDepend\Source\AST\ASTAnonymousClass given, called in /home/j/workspace/FundraisingFrontend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php on line 214 and defined in /home/j/workspace/FundraisingFrontend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/AST/AbstractASTNode.php:367
Stack trace:
#0 /home/j/workspace/FundraisingFrontend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php(214): PDepend\Source\AST\AbstractASTNode->addChild(Object(PDepend\Source\AST\ASTAnonymousClass))
#1 /home/j/workspace/FundraisingFrontend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php(188): PDepend\Source\Language\PHP\PHPParserVersion70->parseAnonymousClassDeclaration(Object(PDepend\Source\AST\ASTAllocationExpression))
#2 /home/j/workspace/FundraisingFrontend/vend in /home/j/workspace/FundraisingFrontend/vendor/pdepend/pdepend/src/main/php/PDepend/Source/AST/AbstractASTNode.php on line 367

```

We get this error when running PHPMD against our tests. Not when running against the production code. Presumably this is some PHP7 problem.

**Comment by marcinlawnik** [comment]
I also encountered this issue:

    PHP 7.0.5 (cli) (built: Mar 30 2016 10:02:33) ( ZTS )
    Copyright (c) 1997-2016 The PHP Group
    Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies

`Windows 10
`

**Comment by BorislavSabev** [comment]
Hello.

I also have the same issue but under 5.6:

```
PHP 5.6.24-0+deb8u1 (cli) (built: Jul 26 2016 08:17:07) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
    with Xdebug v2.2.5, Copyright (c) 2002-2014, by Derick Rethans
```

It seems to me that this happens when parsing some odd file but I cannot coin it down to an exact php file. Some tips on how to find which file it is?

Error output:
```
PHP Catchable fatal error:  Argument 1 passed to PDepend\Source\AST\AbstractASTNode::addChild() must implement interface PDepend\Source\AST\ASTNode, instance of PDepend\Source\AST\ASTAnonymousClass given, called in /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php on line 214 and defined in /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/AST/AbstractASTNode.php on line 367
PHP Stack trace:
PHP   1. {main}() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/bin/pdepend:0
PHP   2. PDepend\TextUI\Command::main() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/bin/pdepend:72
PHP   3. PDepend\TextUI\Command->run() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/TextUI/Command.php:627
PHP   4. PDepend\TextUI\Runner->run() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/TextUI/Command.php:243
PHP   5. PDepend\Engine->analyze() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/TextUI/Runner.php:314
PHP   6. PDepend\Engine->performParseProcess() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php:319
PHP   7. PDepend\Source\Language\PHP\AbstractPHPParser->parse() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php:571
PHP   8. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalStatement() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:382
PHP   9. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:5682
PHP  10. PDepend\Source\Language\PHP\AbstractPHPParser->parseVariableOrConstantOrPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2610
PHP  11. PDepend\Source\Language\PHP\AbstractPHPParser->parseVariableOrFunctionPostfixOrMemberPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4226
PHP  12. PDepend\Source\Language\PHP\AbstractPHPParser->parseMemberPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4280
PHP  13. PDepend\Source\Language\PHP\AbstractPHPParser->parseMethodOrPropertyPostfix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:3952
PHP  14. PDepend\Source\Language\PHP\AbstractPHPParser->parseMethodPostfix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4086
PHP  15. PDepend\Source\Language\PHP\AbstractPHPParser->parseArguments() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4169
PHP  16. PDepend\Source\Language\PHP\AbstractPHPParser->parseExpressionList() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4205
PHP  17. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2521
PHP  18. PDepend\Source\Language\PHP\AbstractPHPParser->parseAllocationExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2626
PHP  19. PDepend\Source\Language\PHP\PHPParserVersion70->parseAllocationExpressionTypeReference() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:1475
PHP  20. PDepend\Source\Language\PHP\PHPParserVersion70->parseAnonymousClassDeclaration() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:188
PHP  21. PDepend\Source\AST\AbstractASTNode->addChild() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:214
```


I am also getting the same error when running phpmd on the same code:
```
PHP Catchable fatal error:  Argument 1 passed to PDepend\Source\AST\AbstractASTNode::addChild() must implement interface PDepend\Source\AST\ASTNode, instance of PDepend\Source\AST\ASTAnonymousClass given, called in /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php on line 214 and defined in /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/AST/AbstractASTNode.php on line 367
PHP Stack trace:
PHP   1. {main}() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/bin/phpmd:0
PHP   2. PHPMD\TextUI\Command::main() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/bin/phpmd:122
PHP   3. PHPMD\TextUI\Command->run() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php:170
PHP   4. PHPMD\PHPMD->processFiles() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php:130
PHP   5. PHPMD\Parser->parse() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/PHPMD.php:222
PHP   6. PDepend\Engine->analyze() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/Parser.php:123
PHP   7. PDepend\Engine->performParseProcess() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php:319
PHP   8. PDepend\Source\Language\PHP\AbstractPHPParser->parse() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php:571
PHP   9. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalStatement() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:382
PHP  10. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:5682
PHP  11. PDepend\Source\Language\PHP\AbstractPHPParser->parseVariableOrConstantOrPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2610
PHP  12. PDepend\Source\Language\PHP\AbstractPHPParser->parseVariableOrFunctionPostfixOrMemberPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4226
PHP  13. PDepend\Source\Language\PHP\AbstractPHPParser->parseMemberPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4280
PHP  14. PDepend\Source\Language\PHP\AbstractPHPParser->parseMethodOrPropertyPostfix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:3952
PHP  15. PDepend\Source\Language\PHP\AbstractPHPParser->parseMethodPostfix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4086
PHP  16. PDepend\Source\Language\PHP\AbstractPHPParser->parseArguments() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4169
PHP  17. PDepend\Source\Language\PHP\AbstractPHPParser->parseExpressionList() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4205
PHP  18. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2521
PHP  19. PDepend\Source\Language\PHP\AbstractPHPParser->parseAllocationExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2626
PHP  20. PDepend\Source\Language\PHP\PHPParserVersion70->parseAllocationExpressionTypeReference() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:1475
PHP  21. PDepend\Source\Language\PHP\PHPParserVersion70->parseAnonymousClassDeclaration() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:188
PHP  22. PDepend\Source\AST\AbstractASTNode->addChild() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:214
PHP Catchable fatal error:  Argument 1 passed to PDepend\Source\AST\AbstractASTNode::addChild() must implement interface PDepend\Source\AST\ASTNode, instance of PDepend\Source\AST\ASTAnonymousClass given, called in /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php on line 214 and defined in /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/AST/AbstractASTNode.php on line 367
PHP Stack trace:
PHP   1. {main}() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/bin/phpmd:0
PHP   2. PHPMD\TextUI\Command::main() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/bin/phpmd:122
PHP   3. PHPMD\TextUI\Command->run() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php:170
PHP   4. PHPMD\PHPMD->processFiles() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php:130
PHP   5. PHPMD\Parser->parse() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/PHPMD.php:222
PHP   6. PDepend\Engine->analyze() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/Parser.php:123
PHP   7. PDepend\Engine->performParseProcess() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php:319
PHP   8. PDepend\Source\Language\PHP\AbstractPHPParser->parse() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php:571
PHP   9. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalStatement() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:382
PHP  10. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:5682
PHP  11. PDepend\Source\Language\PHP\AbstractPHPParser->parseVariableOrConstantOrPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2610
PHP  12. PDepend\Source\Language\PHP\AbstractPHPParser->parseVariableOrFunctionPostfixOrMemberPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4226
PHP  13. PDepend\Source\Language\PHP\AbstractPHPParser->parseMemberPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4280
PHP  14. PDepend\Source\Language\PHP\AbstractPHPParser->parseMethodOrPropertyPostfix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:3952
PHP  15. PDepend\Source\Language\PHP\AbstractPHPParser->parseMethodPostfix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4086
PHP  16. PDepend\Source\Language\PHP\AbstractPHPParser->parseArguments() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4169
PHP  17. PDepend\Source\Language\PHP\AbstractPHPParser->parseExpressionList() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4205
PHP  18. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2521
PHP  19. PDepend\Source\Language\PHP\AbstractPHPParser->parseAllocationExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2626
PHP  20. PDepend\Source\Language\PHP\PHPParserVersion70->parseAllocationExpressionTypeReference() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:1475
PHP  21. PDepend\Source\Language\PHP\PHPParserVersion70->parseAnonymousClassDeclaration() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:188
PHP  22. PDepend\Source\AST\AbstractASTNode->addChild() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:214
PHP Catchable fatal error:  Argument 1 passed to PDepend\Source\AST\AbstractASTNode::addChild() must implement interface PDepend\Source\AST\ASTNode, instance of PDepend\Source\AST\ASTAnonymousClass given, called in /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php on line 214 and defined in /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/AST/AbstractASTNode.php on line 367
PHP Stack trace:
PHP   1. {main}() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/bin/phpmd:0
PHP   2. PHPMD\TextUI\Command::main() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/bin/phpmd:122
PHP   3. PHPMD\TextUI\Command->run() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php:170
PHP   4. PHPMD\PHPMD->processFiles() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php:130
PHP   5. PHPMD\Parser->parse() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/PHPMD.php:222
PHP   6. PDepend\Engine->analyze() /home/borislav/Workspaces/ProjectName/library/vendor/phpmd/phpmd/src/main/php/PHPMD/Parser.php:123
PHP   7. PDepend\Engine->performParseProcess() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php:319
PHP   8. PDepend\Source\Language\PHP\AbstractPHPParser->parse() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php:571
PHP   9. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalStatement() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:382
PHP  10. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:5682
PHP  11. PDepend\Source\Language\PHP\AbstractPHPParser->parseVariableOrConstantOrPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2610
PHP  12. PDepend\Source\Language\PHP\AbstractPHPParser->parseVariableOrFunctionPostfixOrMemberPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4226
PHP  13. PDepend\Source\Language\PHP\AbstractPHPParser->parseMemberPrimaryPrefix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4280
PHP  14. PDepend\Source\Language\PHP\AbstractPHPParser->parseMethodOrPropertyPostfix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:3952
PHP  15. PDepend\Source\Language\PHP\AbstractPHPParser->parseMethodPostfix() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4086
PHP  16. PDepend\Source\Language\PHP\AbstractPHPParser->parseArguments() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4169
PHP  17. PDepend\Source\Language\PHP\AbstractPHPParser->parseExpressionList() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:4205
PHP  18. PDepend\Source\Language\PHP\AbstractPHPParser->parseOptionalExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2521
PHP  19. PDepend\Source\Language\PHP\AbstractPHPParser->parseAllocationExpression() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:2626
PHP  20. PDepend\Source\Language\PHP\PHPParserVersion70->parseAllocationExpressionTypeReference() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php:1475
PHP  21. PDepend\Source\Language\PHP\PHPParserVersion70->parseAnonymousClassDeclaration() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:188
PHP  22. PDepend\Source\AST\AbstractASTNode->addChild() /home/borislav/Workspaces/ProjectName/library/vendor/pdepend/pdepend/src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:214
```

Any information on how to investigate more and add more information would be greatly appreciated.

**Comment by MatissJanis** [comment]
I experienced the same issue because I was using anonymous classes.

```
$someVar = new class {
    function someMethod()
    {
        return true;
    }
}
```

**Comment by Fjarlaegur** [comment]
Same here. Using jenkins to test an untouched Laravel project (Except for the build files ofcourse).

**Comment by jimdelois** [comment]
Duplicate of https://github.com/pdepend/pdepend/issues/250

**Comment by manuelpichler** [comment]
Resolved in 2111906
